File: app/src/main/java/com/kickstart/application/di/module/FragmentModule.java
package com.kickstart.application.di.module;

import com.kickstart.application.di.scope.PerActivity;
import com.kickstart.application.ui.BaseFragment;

import dagger.Module;
import dagger.Provides;

/**
 * A module to expose the objects in the fragment state to fragment-level
 * object graph.
 */
@Module
public class FragmentModule {
    private final BaseFragment mFragment;

    public FragmentModule(BaseFragment fragment) {
        mFragment = fragment;
    }

    @Provides
    @PerActivity
    BaseFragment provideFragment() {
        return mFragment;
    }
}

File: app/src/main/java/com/kickstart/railway/di/module/FragmentModule.kt
package com.kickstart.railway.di.module

import com.kickstart.railway.di.scope.PerActivity
import com.kickstart.railway.ui.BaseFragment

import dagger.Module
import dagger.Provides

/**
 * A module to expose the objects in the fragment state to fragment-level
 * object graph.
 */
@Module
class FragmentModule(private val mFragment: BaseFragment) {

    @Provides
    @PerActivity
    internal fun provideFragment(): BaseFragment {
        return mFragment
    }
}

